docs/blog/pydantic-v2.md

```diff
@@ -0,0 +1,221 @@
+* adding validation to `dataclass`
+* validating URL arguments, query strings, headers, etc. in FastAPI
+* adding validation to `TypedDict`
+* function argument validation
+
+Basically anywhere were you don't care about a traditional model class.
+
+### Strict API & API documentation :smile:
+
+When preparing a pydantic V2, we'll make a strict distinction between the public API and private functions & classes.
+Private objects clearly identified as private via `_internal` sub package to discourage use.
+
+The public API will have API documentation. I've recently been working with the wonderful
+[mkdocstrings](https://github.com/mkdocstrings/mkdocstrings) for both 
+[dirty-equals](https://dirty-equals.helpmanual.io/) and
+[watchfiles](https:://watchfiles.helpmanual.io/) documentation. I intend to use `mkdocstrings` to generate complete
+API documentation.
+
+This wouldn't replace the current example-based somewhat informal documentation style byt instead will augment it.
+
+### No pure python implementation :frowning:
+
+Since pydantic-core is written in Rust, and I have absolutely no intention of rewriting it in python,
+pydantic V2 will only work where a binary package can be installed.
+
+pydantic-core will provide binaries in PyPI for (at least):
+
+* **Linux**: `x86_64`, `aarch64`, `i686`, `armv7l`, `musl-x86_64` & `musl-aarch64`
+* **MacOS**: `x86_64` & `arm64` (except python 3.7)
+* **Windows**: `amd64` & `win32`
+* **Web Assembly**: `wasm32` (pydantic-core is already compiled for wasm32 using emscripten and unit tests pass, 
+  except where cpython itself has [problems](https://github.com/pyodide/pyodide/issues/2841))
+
+Other binaries can be added provided they can be (cross-)compiled on github actions.
+If no binary is available from PyPI, pydantic-core can be compiled from source if Rust stable is available.
+
+The only place where I know this will cause problems is Raspberry Pi, which is a bit of 
+[mess](https://github.com/piwheels/packages/issues/254) when it comes to packages written in rust for python.
+Effectively, until that's fixed you'll likely have to install pydantic-core with 
+`pip install -i https://pypi.org/simple/ pydantic-core`.
+
+### Pydantic becomes a pure python package :confused:
+
+Pydantic v1.X is a pure python code base but is compiled with cython to provide some performance improvements.
+Since the "hot" code is moved to pydantic-core, pydantic itself can go back to being a pure python package.
+
+This should significantly reduce the size of the pydantic package and make unit tests of pydantic much faster.
+In addition, some constraints on pydantic code can be removed once it no-longer has to be compilable with cython.
+
+Some pieces of edge logic could get a little slower as they're no longer compiled.
+
+## Other Improvements :smile:
+
+* Recursive models
+* Documentation examples you can edit and run
+
+TODO
+
+## Removed Features :neutral_face:
+
+* `__root__` models
+
+TODO
+
+## Conversion Table
+
+TODO
```

Comment:
Consider adding an ending section / wrapup. The blog just trails off with these features so either add a short sum-up at the end or have some conclusions *before* this list of features and proposed changes and say see below for the full list. 

Given folks will have lots of opinions about these changes, it could be worth getting ahead of that and explicitly stating how you would like folks to contribute thoughts, opinions etc. E.g. an issue thread?

Reply:
done.